### PR TITLE
fix version

### PR DESCRIPTION
**Pull request by yaohuan771634**

### CODE REVIEW

File: smart-common/src/main/java/org/smartdata/versioninfo/VersionInfoWrite.java [hunk 1]
```diff
@@ -70,18 +78,15 @@ private String getBuildTime() {
...
-    BufferedReader br = new BufferedReader(fileReader);
-    while ((st = br.readLine()) != null) {
-      if (st.contains("<version>")) {
-        return st.trim().substring("<version>".length(),
-            st.trim().length() - "</version>".length());
-      }
-    }
-    return "Not found";
+  private String getVersionInfo(String fileName) throws Exception {
+    DocumentBuilderFactory dbFactory = DocumentBuilderFactory.newInstance();
```
Comment (PHILO-HE) [inline smart-common/src/main/java/org/smartdata/versioninfo/VersionInfoWrite.java]: docBuilderFactory

File: smart-common/src/main/java/org/smartdata/versioninfo/VersionInfoWrite.java [hunk 2]
```diff
@@ -70,18 +78,15 @@ private String getBuildTime() {
...
+    DocumentBuilder dbBuilder = dbFactory.newDocumentBuilder();
```
Comment (PHILO-HE) [inline smart-common/src/main/java/org/smartdata/versioninfo/VersionInfoWrite.java]: docBuilder

File: smart-common/src/main/java/org/smartdata/versioninfo/VersionInfoWrite.java [hunk 3]
```diff
@@ -70,18 +78,15 @@ private String getBuildTime() {
...
+    Document doc = dbBuilder.parse(fileName);
+    NodeList nList = doc.getElementsByTagName("parent");
```
Comment (PHILO-HE) [inline smart-common/src/main/java/org/smartdata/versioninfo/VersionInfoWrite.java]: parentList

File: smart-common/src/main/java/org/smartdata/versioninfo/VersionInfoWrite.java [hunk 4]
```diff
@@ -70,18 +78,15 @@ private String getBuildTime() {
...
+    Element parent = (Element) nList.item(0);
+    NodeList cList = parent.getElementsByTagName("version");
```
Comment (PHILO-HE) [inline smart-common/src/main/java/org/smartdata/versioninfo/VersionInfoWrite.java]: versionList
if versionList == null, return "Not Found".

File: smart-common/src/main/java/org/smartdata/versioninfo/VersionInfoWrite.java [hunk 5]
```diff
@@ -36,7 +42,7 @@
 
 public class VersionInfoWrite {
   File directory = new File("");
-  String pom = directory.getAbsolutePath() + "/" + "pom.xml";
+  String pom = directory.getAbsolutePath() + "/smart-common/pom.xml";
```
Comment (PHILO-HE) [inline smart-common/src/main/java/org/smartdata/versioninfo/VersionInfoWrite.java]: The old code use project pom not smart-commont code? If the project can be read, maybe using it is better.

Comment (yaohuan771634) [reply smart-common/src/main/java/org/smartdata/versioninfo/VersionInfoWrite.java]: in project pom, tag <version>'s parent tag is <project>,the root node. 
it is better use to ssm-common pom.

**Pull request merged**